feat: support path and relative-path  prop

hairyf opened the pull request:
close #24 

<!-- insert PR description above -->
-----
<a href="https://stackblitz.com/~/github.com/hairyf/vitepress-demo-preview/tree/hairyf%2Fpatch-82373"><img src="https://developer.stackblitz.com/img/review_pr_small.svg" alt="Review PR in StackBlitz Codeflow" align="left" width="103" height="20"></a> _Submitted with [StackBlitz Codeflow](https://stackblitz.com/~/github.com/hairyf/vitepress-demo-preview/tree/hairyf%2Fpatch-82373)._

flingyp commented:
Jump to github demo. [@vitepress-demo-preview/component](https://github.com/flingyp/vitepress-demo-preview/blob/main/packages/component) package should achieve for the function. Right?

hairyf commented:
of course! I'll give it a try

hairyf commented:
Additionally, I haven't found any test cases to verify this functionality. The dependencies in the example are linked to remote addresses. How should I proceed with testing?

flingyp commented:
OK, but the codes you modified doesn't do the function. **relativePath** should is a online link url but you get the absolute path of the local component.

flingyp commented:
We should make sure that a **relativePath** prop is resolvable by the plugin. Right?
Maybe you can do it in the [@vitepress-demo-preview/component](https://github.com/flingyp/vitepress-demo-preview/blob/main/packages/component) package and provide the function to jump the link.

hairyf commented:
<script lang="ts" setup>
const props = defineProps<{
  relativePath: string
}>()

const githubPath = `https://www.github.com/.../tree/main${props.relativePath}`
</script>

<template>
  <!-- .... -->
  <a :href="githubPath">to github edit</a>
</template>

hairyf commented:
Translation: I added support for that feature in [markdown-it-vitepress-demo#metadata](https://github.com/hairyf/markdown-it-vitepress-demo) and implemented other features as well. It has indeed been helpful to me. Additionally, I would like to thank the project for providing me with great inspiration!

flingyp commented:
Thanks, your attention and contribution!

I can test the feature at example and this is the result I saw. You only need to custom a `demo-preview` component and the usage remains unchanged.
![image](https://github.com/flingyp/vitepress-demo-preview/assets/57666140/3bcafb8a-b820-4147-b4b5-ec1edf1ce36f)

If this meets your expectations, I will merge this PR.

hairyf commented:
yes, however, it should replace `\` with`/`

flingyp commented:
> yes, however, it should replace `\` with`/`

Yeah, well, you can change the code and submit it. I will merge the pr. Then you just need to customize your own preview component. Thanks!
